src/main/java/com/fasterxml/jackson/databind/deser/std/StringCollectionDeserializer.java

```diff
@@ -304,6 +306,21 @@ public Object deserializeWithType(JsonParser p, DeserializationContext ctxt,
                 throw JsonMappingException.wrapWithPath(e, result, result.size());
             }
         }
+        if (value != null) {
```

Comment:
Hmmmh. What kind of value are we expecting here? `JsonToken.VALUE_STRING` was handled earlier wasn't it?

Reply:
- The method is invoked when the token is *not* `START_ARRAY`
- This code path is visited when `ACCEPT_SINGLE_VALUE_AS_ARRAY` is *enabled*

So the token here could be anything non-array that is supported by the `valueDes`. The `value` could then be null.

Other than the `null` check this code is copied from `CollectionDeserializer`. Though in that case it does these checks *before* going through the `valueDes` (it just does them on `parser.getText()` if the token is `VALUE_STRING`). Is that a better approach?

Reply:
This to me just seems wrong... values returned by a deserializer (or `getNullValue()`) really should not be post-processed by logic like this: coercion, in particular, is to be applied to incoming tokens and not on something deserializer has already decoded.

So this just feels conceptually against design. I understand it may result in desired outcome, it just does not seem to get there in a way API should work.

Not sure how to proceed here to be honest.

Reply:
I'll take a look how CollectionDeserializer does it, I don't think it processes the getNullValue. Maybe I can replicate that.